app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -2325,41 +2401,79 @@ class BrowserTabFragment :
+
+                val isFavorite = viewState.favorite != null
+                addFavoriteMenuItem?.isVisible = viewState.addFavorite.isEnabled()
+                addFavoriteMenuItem.label {
+                    when {
+                        viewState.addFavorite.isHighlighted() -> getString(R.string.addFavoriteMenuTitleHighlighted)
+                        isFavorite -> getString(R.string.removeFavoriteMenuTitle)
+                        else -> getString(R.string.addFavoriteMenuTitle)
+                    }
                 }
-                fireproofWebsitePopupMenuItem?.isEnabled = viewState.canFireproofSite
-                fireproofWebsitePopupMenuItem?.isChecked = viewState.canFireproofSite && viewState.isFireproofWebsite
-                sharePageMenuItem?.isEnabled = viewState.canSharePage
-                whitelistPopupMenuItem?.isEnabled = viewState.canWhitelist
-                whitelistPopupMenuItem?.text =
-                    getText(if (viewState.isWhitelisted) R.string.enablePrivacyProtection else R.string.disablePrivacyProtection)
-                brokenSitePopupMenuItem?.isEnabled = viewState.canReportSite
-                requestDesktopSiteCheckMenuItem?.isEnabled = viewState.canChangeBrowsingMode
-                requestDesktopSiteCheckMenuItem?.isChecked = viewState.isDesktopBrowsingMode
+                addFavoriteMenuItem.setIcon(if (isFavorite) drawable.ic_favorite_solid_16 else drawable.ic_favorite_16)
 
-                newEmailAliasMenuItem?.let {
-                    it.visibility = if (viewState.isEmailSignedIn) VISIBLE else GONE
+                fireproofWebsiteMenuItem?.isVisible = viewState.canFireproofSite
+                fireproofWebsiteMenuItem?.label {
+                    getString(
+                        if (viewState.isFireproofWebsite) {
+                            R.string.fireproofWebsiteMenuTitleRemove
+                        } else {
+                            R.string.fireproofWebsiteMenuTitleAdd
+                        }
+                    )
                 }
+                fireproofWebsiteMenuItem?.setIcon(if (viewState.isFireproofWebsite) drawable.ic_fire_16 else drawable.ic_fireproofed_16)
+
+                createAliasMenuItem?.isVisible = viewState.isEmailSignedIn
 
-                addToHome?.let {
-                    it.visibility = if (viewState.addToHomeVisible) VISIBLE else GONE
-                    it.isEnabled = viewState.addToHomeEnabled
+                changeBrowserModeMenuItem?.isVisible = viewState.canChangeBrowsingMode
+                changeBrowserModeMenuItem.label {
+                    getString(
+                        if (viewState.isDesktopBrowsingMode) {
+                            R.string.requestMobileSiteMenuTitle
+                        } else {
+                            R.string.requestDesktopSiteMenuTitle
+                        }
+                    )
                 }
+                changeBrowserModeMenuItem?.setIcon(
```

Comment:
Let's discuss pros/cons. I think is a good approach and it will work, but at the same time feels not right. It will work because menu items have 2 possible states, but not if we introduce more. Text will still be changed by another logic. And changing selected attr to change the icon, but knowing the view is not actually "selected" can be confusing in the future.